### PR TITLE
Simple subpackage support for cython_freeze

### PR DESCRIPTION
**Pull request by rjungbeck**
The PR discussed in https://github.com/cython/cython/issues/4852

### CODE REVIEW

File: bin/cython_freeze [hunk 1]
```diff
@@ -51,6 +64,23 @@ print("""\
...
+    static PyModuleDef %sModule = {
+        PyModuleDef_HEAD_INIT, "%s", NULL, -1, NULL,
+        NULL, NULL, NULL, NULL
+        };
+
+    static PyObject * PyInit_%s(void)
+    {
+        PyObject * mod = PyModule_Create( & %sModule);
+        PyModule_AddObject(mod, "__path__", Py_BuildValue("()"));
+        return mod;
```
Comment (scoder) [inline bin/cython_freeze:79]: The generated C code lacks error handling everywhere. Could you please add that?

File: bin/cython_freeze [hunk 2]
```diff
@@ -29,7 +29,20 @@ if options.output:
...
     sys.stdout = open(options.output, 'w')
 
-modules = [basename(splitext(x)[0]).replace('.', '_') for x in args]
+packages=[]
+modules=[]
+fullNames=[]
+
+for x in args:
+    moduleName=basename(splitext(x)[0]).replace('.', '_')
+    fullName=splitext(x)[0].replace(".", "_").replace("/", ".")
```
Comment (scoder) [inline bin/cython_freeze:38]: I think it's worth giving a name to `splitext(x)[0]`. And also to `x`, e.g. `file_name`.

**Review by scoder — COMMENTED** [review]
Thanks. Please adapt the code style to the rest of the file (which mostly follows PEP-8), e.g. spaces around operators, variable names with underscores.

File: bin/cython_freeze (hunk 1 above)
Comment (rjungbeck) [reply bin/cython_freeze]: Changes were made.

File: bin/cython_freeze (hunk 2 above)
Comment (rjungbeck) [reply bin/cython_freeze]: Chngnes were made

File: bin/cython_freeze [hunk 3]
```diff
@@ -51,6 +64,23 @@ print("""\
...
+    
+    static PyModuleDef %sModule = {
+        PyModuleDef_HEAD_INIT, "%s", NULL, -1, NULL,
+        NULL, NULL, NULL, NULL
+        };
+
+    static PyObject * PyInit_%s(void)
+    {
+        PyObject * mod = PyModule_Create( & %sModule);
+        PyModule_AddObject(mod, "__path__", Py_BuildValue("()"));
```
[the remark this answers is not in the document]
Comment (rjungbeck) [reply bin/cython_freeze]: Changes were made

File: bin/cython_freeze [hunk 4]
```diff
@@ -51,6 +65,27 @@ print("""\
 for name in modules:
     print("PyMODINIT_FUNC MODINIT(%s) (void);" % name)
 
+for package in packages:
+    print('''
+    
+    static PyModuleDef %sModule = {
+        PyModuleDef_HEAD_INIT, "%s", NULL, -1, NULL,
+        NULL, NULL, NULL, NULL
+        };
```
Comment (scoder) [inline bin/cython_freeze:74]:
```suggestion
    };
```

File: bin/cython_freeze [hunk 5]
```diff
@@ -51,6 +65,27 @@ print("""\
...
+    static PyObject * PyInit_%s(void)
+    {
+        PyObject * mod = PyModule_Create( & %sModule);
+        if (!mod) {
+          fprintf(stderr, "Error creating module %s");
+          exit(1);
+          }
+        PyModule_AddStringConstant(mod, "__path__", "%s");
+        return mod;
+        }
```
Comment (scoder) [inline bin/cython_freeze:85]:
```suggestion
    }
```

File: bin/cython_freeze [hunk 6]
```diff
@@ -51,6 +65,27 @@ print("""\
...
 
+for package in packages:
+    print('''
+    
+    static PyModuleDef %sModule = {
+        PyModuleDef_HEAD_INIT, "%s", NULL, -1, NULL,
+        NULL, NULL, NULL, NULL
+        };
+
+    static PyObject * PyInit_%s(void)
```
Comment (scoder) [inline bin/cython_freeze:76]: There are macros for this function name and its declaration, see above.

**Pull request closed without merging**